schedview/app/prenight/prenight.py

```diff
@@ -932,7 +947,16 @@ def prenight_app(*args, **kwargs):
         except KeyError:
             data_dir = None
 
-        prenight = RestrictedInputPrenight(data_dir=data_dir)
+        specified_data_files = {}
+        data_args = set(["opsim_db", "scheduler", "rewards"]) & set(kwargs.keys())
+        for data_arg in data_args:
+            # Fully resolve the path to the file.
+            file_path = Path(kwargs[data_arg]).resolve()
+            kwargs[data_arg] = str(file_path)
+            if data_arg in kwargs:
+                specified_data_files[data_arg] = str(file_path)
+
+        prenight = RestrictedInputPrenight(data_dir=data_dir, **specified_data_files)
```

Comment:
Oh interesting .. so I think this means that if you do not allow arbitrary data urls (which I thought meant uploading files only, rather than files from the filesystem), that you have to pre-specify which data directories should be allowed? 
So the choices are -- data files from paths that are specified at startup vs. any data file + URL uploads? 
(I'm not sure if that would be clear to other people, but wasn't clear to me, if that's how it works)

Reply:
Yes, that's right.